compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -1765,7 +1765,20 @@ impl<'a> Parser<'a> {
     /// Parses an `if` expression (`if` token already eaten).
     fn parse_if_expr(&mut self, attrs: AttrVec) -> PResult<'a, P<Expr>> {
         let lo = self.prev_token.span;
-        let cond = self.parse_cond_expr()?;
+        let mut cond = self.parse_cond_expr()?.into_inner();
```

Comment:
Is the `into_inner` necessary? IIUC, you need that to be able to reassing, but wouldn't changing line 1777 to be `cond = P(cond.peel_parens_owned());` work as well and make this "unwrapping/rewrapping" unnecessary in the happy path? I'm just concerned about doing unnecessary work that might marginally slow down compile times.